ext/nnstreamer/tensor_decoder/tensordec-imagesegment.c

```diff
@@ -0,0 +1,350 @@
+/**
+ * GStreamer / NNStreamer tensor_decoder subplugin, "image segment"
+ * Copyright (C) 2019 Jihoon Lee <ulla4571@gmail.com>
+ * Copyright (C) 2019 niklasjang <niklasjang@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensordec-directvideo.c
+ * @date	19 Oct 2019
+ * @brief	NNStreamer tensor-decoder subplugin, "image segment",
+ *              which detects objects and paints their regions.
+ *
+ * @see		https://github.com/nnsuite/nnstreamer
+ * @author	Jihoon Lee <ulla4571@gmail.com>
+ *          niklasjang <niklasjang@gmail.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * option1: Decoder mode of image segmentation
+ *          Available : tflite-deeplab
+ * option2: Video Output Dimension (WIDTH:HEIGHT)
+ *          This depends on option1 values
+ * option3: Video Input Dimension (WIDHT: HEIGHT)
+ *          This depends on option1 values
+ */
+
+#include <string.h>
+#include <glib.h>
+#include <gst/video/video-format.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
+
+void init_is (void) __attribute__ ((constructor));
+void fini_is (void) __attribute__ ((destructor));
+
+#define TFLITE_DEEPLAB_TOTAL_LABELS    21
+#define TFLITE_DEEPLAB_IMAGE_WIDTH     257
+#define TFLITE_DEEPLAB_IMAGE_HEIGHT    257
+
+/**
+ * @brief There can be different schemes for image segmentation.
+ */
+typedef enum
+{
+  TFLITE_DEEPLAB_IMAGE_SEGMENT = 0,
+  UNKNOWN_IMAGE_SEGMENT,
+} image_segment_modes;
+
+/**
+ * @brief List of image-segmentation decoding schemes in string
+ */
+static const char *is_modes[] = {
+  [TFLITE_DEEPLAB_IMAGE_SEGMENT] = "tflite-deeplab",
+  NULL,
+};
+
+/**
+ * @brief Data structure for image segmentation info.
+ */
+typedef struct
+{
+  image_segment_modes mode; /**< The image segmentation decoding mode */
+  guint **segment_map;  /**< The image segmentated map */
+
+  /*  From option2 */
+  guint o_width; /**< Output video Width */
```

Comment:
NOTE THAT THIS OPTION IS ONLY FOR TFLITE_DEEPLAB_IMAGE_SEGMENT!

Reply:
Sry, I don't know what to do for it

Reply:
Does it mean do not place the option dependent value in common data structure?